flow/src/main/kotlin/org/testing/goody/GoodyOps.kt

```diff
@@ -0,0 +1,94 @@
+package org.testing.goody
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.core.contracts.*
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import net.corda.core.node.ServiceHub
+import net.corda.core.transactions.TransactionBuilder
+import org.testing.goody.contracts.Candy
+import org.testing.goody.contracts.Goody
+import org.testing.goody.flows.InsufficientGoodiesException
+import java.security.PublicKey
+
+object GoodyOps {
+    private fun deriveState(txState: TransactionState<Goody.State>, amount: Amount<Issued<Candy>>, owner: AbstractParty)
+        = txState.copy(data = txState.data.copy(amount = amount, owner = owner))
+
+    fun generateIssue(tx: TransactionBuilder, amount: Amount<Issued<Candy>>, owner: AbstractParty, notary: Party)
+        = generateIssue(tx, TransactionState(Goody.State(owner, amount), Goody.PROGRAM_ID, notary), Goody.Commands.Issue())
+
+    fun generateIssue(tx: TransactionBuilder,
+                      transactionState: TransactionState<Goody.State>,
+                      issueCommand: CommandData): Set<PublicKey> {
+        check(tx.inputStates().isEmpty())
+        check(tx.outputStates().map(TransactionState<*>::data).filterIsInstance(transactionState.javaClass).isEmpty())
+        require(transactionState.data.amount.quantity > 0)
+        val at = transactionState.data.amount.token.issuer
+        val commandSigner = at.party.owningKey
+        tx.addOutputState(transactionState)
+        tx.addCommand(issueCommand, commandSigner)
+        return setOf(commandSigner)
+    }
+
+    @Throws(InsufficientGoodiesException::class)
+    @Suspendable
+    fun generateTransfer(services: ServiceHub,
```

Comment:
Is there no way to re-use the existing generateSpend support in finance for this?

Reply:
I have tried to implement a `FungibleAsset` that isn't `Cash`, on the assumption that this is a reasonable thing to do. The `generateSpend` in `:finance` appears to be _specific_ to `Amount<Currency>`, and is therefore unsuitable. (Although I've obviously based my implementation upon it.)